app/controllers/event_api_controller.rb

```diff
@@ -0,0 +1,29 @@
+class EventApiController < ApplicationController
+
+  def index
+    if event_params.include?(:time_period)
+      @events = events_by_time_period
+      render json: @events, status: :created
+    else
+      render json: { error: 'Valid time period required.' }, 
+        status: :unprocessable_entity
+    end
+  end
+
+  private
+    def event_params
+      params.permit(:time_period)
+    end
+
+    def events_by_time_period
+      case event_params[:time_period]
+      when 'upcoming'
+        Event.where("Date(start_time) >= ?", Date.today)
+      when 'past'
+        Event.where("Date(start_time) < ?", Date.today)
```

Comment:
I've never seen a `Date()` cast like this before. Unless `start_time` is stored as a `String`, you _shouldn't_ need it.